libs/events/events_test.go

```diff
@@ -196,11 +179,27 @@ func TestAddAndRemoveListenerConcurrency(t *testing.T) {
 	}
 	defer evsw.Stop()
 
-	done1 := make(chan uint64)
-	done2 := make(chan uint64)
+	done1 := make(chan struct{})
+	done2 := make(chan struct{})
+
+	go func() {
+		for i := 0; i < roundCount; i++ {
+			evsw.RemoveListener("listener")
+		}
+		done1 <- struct{}{}
```

Comment:
`close(done1)`

Reply:
here chan can recycled automatically by the garbage collector， but I'll close it explicit